lib/modules.nix

```diff
@@ -115,8 +115,21 @@ rec {
 
       checkUnmatched =
         if config._module.check && config._module.freeformType == null && merged.unmatchedDefns != [] then
-          let inherit (head merged.unmatchedDefns) file prefix;
-          in throw "The option `${showOption prefix}' defined in `${file}' does not exist."
+          let
+            inherit (head merged.unmatchedDefns) file prefix;
+            filteredSet = builtins.removeAttrs (attrByPath (sublist 0 (length prefix - 1) prefix) {} options)
+              [ "_definedNames" "_module" ];
```

Comment:
What's the reason behind `attrByPath (sublist 0 (length prefix - 1) prefix) {} options`?

Also, no need to filter out `_definedNames` anymore

Reply:
> What's the reason behind attrByPath (sublist 0 (length prefix - 1) prefix) {} options?

To retrieve the "parent" of the current option. I'd like to check whether there are any options defined.

> Also, no need to filter out _definedNames anymore

:+1:

Reply:
@Infinisil updated the `removeAttrs` expression.